enhancements/update/targeted-update-edge-blocking.md

```diff
@@ -0,0 +1,247 @@
+Being unable to report Telemetry while still being able to connect to Red Hat's hosted update service seems unlikely; if a cluster can do one (e.g. via a proxy), it should be able to do both.
+Clusters which connect to neither Telemetry nor Red Hat's hosted update service may run a local update service, and that is discussed in [the *divergent Prometheus services* section](#divergent-prometheus-services).
+
+Clusters which opt out of Telemetry are likely still able to connect to the update service.
+When they do, they will be [blocked as cache-misses](#enhanced-graph-data-schema-for-blocking-edges).
+I think that conservative approach is acceptable, because we cannot tell if the non-reporting cluster is vulnerable to the known issues or not.
+Local administrators should instead test their updates locally and make their own decisions about update safety, as discussed in [the *exposing blocking reasons* section](#exposing-blocking-reasons).
+
+#### Exposing blocking reasons
+
+This enhancement provides no mechanism for telling clients if or why a recommended update edge has been blocked, because [the Cincinnati graph format][cincinnati-spec] provides no mechanism for sharing metadata about recommended edges, or even the existence of not-recommended edges.
+Clients might be tempted to want that information to second-guess a graph-data decision to block the edge, but I am [not convinced the information is actionable](overriding-blocked-edges/README.md).
+
+#### Stranding supported clusters
+
+As described [in the documentation][support-documentation], supported updates are still supported even if incoming edges are blocked, and Red Hat will eventually provide supported update paths from any supported release to the latest supported release in its z-stream.
+There is a risk, with the dynamic, per-cluster graph, that targeted edge blocking removes all outgoing update recommendations for some clusters on supported releases.
+The risk is highest for [clusters which are not reporting data](#clusters-not-reporting-data), so an easy way to audit would be to request a graph without specifing an `id` parameter (which [blocks all conditional edges](#enhanced-graph-data-schema-for-blocking-edges)) and to look for old, dead-end releases.
+
+#### Denial of service via ID flooding
+
+Malicious clients may flood an update service with graph requests rotating through a series of nominal `id` parameters in an attempt to consume the update service's cache capacity and/or PromQL volume.
+Update service maintainers should consider rate-limiting requests from a single source, requiring token-based authentication, and other mechanisms for limiting the denial of service exposure of expensive query requests.
+
+## Design Details
+
+### Test Plan
```

Comment:
Either here or in Risks and Mitigations, should we add something about profiling of the proposed promql? ie: at the time that we add a rule it should complete in < 100ms or that rule will not merge? (100ms just an example)

Reply:
> Either here or in Risks and Mitigations, should we add something about profiling of the proposed promql?

Would we have a presubmit job with creds for our internal Telemetry service?  I dunno if that even has a public endpoint...  Or did you see this as "if we get a new `promql` string, we will manually profile it using our personal Telemetry creds before approving the PR", or something?